langs/kotlin.go

```diff
@@ -173,50 +179,120 @@ func (lh *KotlinLangHelper) getFDKAPIVersion() (string, error) {
 		return lh.latestFdkVersion, nil
 	}
 
-	const versionURL = "https://api.bintray.com/search/packages/maven?repo=fnproject&g=com.fnproject.fn&a=fdk"
-	const versionEnv = "FN_JAVA_FDK_VERSION"
-	fetchError := fmt.Errorf("Failed to fetch latest Java FDK javaVersion from %v. Check your network settings or manually override the javaVersion by setting %s", versionURL, versionEnv)
+	const bintrayVersionURL = "https://api.bintray.com/search/packages/maven?repo=fnproject&g=com.fnproject.fn&a=fdk"
+	const mavenVersionUrl = "https://repo1.maven.org/maven2/com/fnproject/fn/fdk/maven-metadata.xml"
 
-	type parsedResponse struct {
-		Version string `json:"latest_version"`
-	}
+	const versionEnv = "FN_JAVA_FDK_VERSION"
+	fetchError := fmt.Errorf("Failed to fetch latest Java FDK javaVersion. Check your network settings or manually override the javaVersion by setting %s", versionEnv)
 	version := os.Getenv(versionEnv)
+
 	if version != "" {
 		return version, nil
 	}
+	version, err := lh.getFDKLastestFromURL(mavenVersionUrl, bintrayVersionURL)
+	if err != nil {
+		return "", fetchError
+	}
 
-	// nishalad95: bin tray TLS certs cause verification issues on OSX, skip TLS verification
+	lh.latestFdkVersion = version
+	return version, nil
+}
+
+func (lh *KotlinLangHelper) getFDKLastestFromURL(comURL string, bintrayURL string) (string, error) {
+	var buf *bytes.Buffer
+	var err error
+	err = fmt.Errorf("All URL failed to respond ")
+
+	//First search for com.fnproject.fn from Maven Central to get the latest version
+	buf, err = lh.getURLResponse(comURL, false)
+	if err == nil {
+		version, e1 := lh.parseMavenResponse(*buf)
+		if e1 == nil {
+			lh.pomType = "maven"
+			return version, e1
+		}
+	}
+
+	//Second time search for com.fnproject.fn from Bintray to get the latest version, if fetch from Maven fails
+	buf, err = lh.getURLResponse(bintrayURL, true)
+	if err == nil {
+		version, e1 := lh.parseBintrayResponse(*buf)
+		if e1 == nil {
+			lh.pomType = "bintray"
+			return version, e1
+		}
+	}
+
+	//In all other case return error as latest FDK version is not identified
+	return "", err
+}
```

Comment:
I see a lot of couple of common methods in both helpers. They are in same package, perhaps we should make them unexported functions in the package and place in Java helper file.

Reply:
DOne